juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule

```diff
@@ -0,0 +1,489 @@
+                description "The \"BCMF CBUF SRAM Protect Parity error\" message reports a hardware parity error detected in the EACHIP memory. This may be due to transient hardware SRAM error on the board.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, clear alarm from CLI ('clear chassis fpc errors fpc-slot <fpc> error-code <code>', where <code> is from 'show chassis alarms').  If multiple logs, reset FPC and monitor.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT002 {
+                event-id EACHIP_EVENT002;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*Double-bit ECC error";
+                description "The \"Double-bit ECC error\" message is caused due to transient Hardware error. On-chip IDMEM memory is ECC protected. If double-bit ECC is detected, a fatal error is raised with level emergency";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "The FPC will restart automatically.  Monitor for recurrence.  Contact JTAC if it occurs again.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT003 {
+                event-id EACHIP_EVENT003;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*DRD.*Protect.*Parity error corrected for alloc state memory ";
+                description "The \"MQSS alloc state memory\" message reports a parity error correction done in the EACHIP state memory. This is due to a software micro-code error or transient hardware error";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, clear alarm from CLI ('clear chassis fpc errors fpc-slot <fpc> error-code <code>', where <code> is from 'show chassis alarms').  If multiple logs, reset FPC and monitor.";
+                }                       
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT004 {
+                event-id EACHIP_EVENT004;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*LI.*Unroll TAIL length overflow";
+                description "The \"Unroll TAIL length overflow\" message reports that the packet size used in the forwarding path at a certain stage is greater than the original packet size. This is likely due to a software micro-code error.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset FPC immediately and monitor for recurrence.  If it recurs, then RMA.";
```

Comment:
Can we use a count based rule here as well?

Reply:
@ConstrainedRandomVar , added trigger based on count